app/config.go

```diff
@@ -27,3 +57,33 @@ var evmosActivators = map[string]func(*vm.JumpTable){
 	"evmos_1": eips.Enable0001,
 	"evmos_2": eips.Enable0002,
 }
+
+// setBaseDenomWithChainID registers the display denom and base denom and sets the
+// base denom for the chain. The function registers different values based on
+// the chainID to allow different configurations in mainnet and testnet.
+func setBaseDenomWithChainID(chainID string) {
+	if utils.IsTestnet(chainID) {
+		if err := sdk.RegisterDenom(types.DisplayDenomTestnet, math.LegacyOneDec()); err != nil {
+			panic(err)
+		}
+		if err := sdk.RegisterDenom(types.BaseDenomTestnet, math.LegacyNewDecWithPrec(1, types.BaseDenomUnit)); err != nil {
+			panic(err)
+		}
+		if err := sdk.SetBaseDenom(types.BaseDenomTestnet); err != nil {
+			panic("can't set base denom")
+		}
+		return
+	}
+
+	if utils.IsMainnet(chainID) {
```

Comment:
@ramacarlucho added this check to panic when the chain ID is not valid

Reply:
@0xstepit we need to be able to support different chain ids that are not testnet nor mainnet, because external teams are using custom chain ids to run their CI